cylc/flow/scripts/cylc.py

```diff
@@ -16,10 +16,36 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """cylc main entry point"""
 
-import argparse
-from contextlib import contextmanager
 import os
 import sys
+
+
+def pythonpath_manip():
+    """Stop PYTHONPATH contaminating the Cylc Environment
+
+    * Remove PYTHONPATH items from sys.path to prevent PYTHONPATH
+      contaminating the Cylc Environment.
+    * Re-add items from CYLC_PYTHONPATH to sys.path.
+
+    See Also:
+        https://github.com/cylc/cylc-flow/issues/5124
+    """
+    if 'CYLC_PYTHONPATH' in os.environ:
+        for item in os.environ['CYLC_PYTHONPATH'].split(os.pathsep):
+            abspath = os.path.abspath(item)
+            sys.path.insert(0, abspath)
+    elif 'PYTHONPATH' in os.environ:
+        os.environ['CYLC_PYTHONPATH'] = os.environ['PYTHONPATH']
```

Comment:
I think this is what should be happening?
```suggestion
    if 'PYTHONPATH' in os.environ:
```

Reply:
Why?

Reply:
I thought it was supposed to be this:
- if PYTHONPATH exists, it might interfere with the Cylc environment, so nuke it
- legit Cylc-specific libs have to be put in CYLC_PYTHONPATH by users instead

Is that right?  If so, this seems wrong:
> No, we definitely want to save the PYTHONPATH as CYLC_PYTHONPATH.

Reply:
i.e.,:
- if PYTHONPATH exists, delete it
- if CYLC_PYTHONPATH exists, put its elements in `sys.path`

(2 ifs, no elsif -- this is what I thought I was suggesting above, in my review comment).